### PR TITLE
feat: Implements the `mongodbatlas_stream_processor` data source

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

- Implements the `streamprocessor` data source schema
- Updates the `pipeline` in the resource_schema to use a string and use validators & planModifiers (see below)
- Refactors `EqualJSON` used in `searchindex` to `common/schemafunc.json` and adds `DiffSuppressJSON`

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/common/schemafunc/json_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package schemafunc_test
+
+import (
+	"testing"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/schemafunc"
+)
+
+func Test_EqualJSON(t *testing.T) {
```
Comment (marcosuma) [inline internal/common/schemafunc/json_test.go:9]: nice!

**Review by marcosuma — APPROVED** [review]

**Review by oarbusi — APPROVED** [review]

File: internal/service/streamprocessor/data_source_schema.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+				Required:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+			},
+			"state": schema.StringAttribute{
+				Computed:            true,
+				Description:         "The state of the stream processor.",
+				MarkdownDescription: "The state of the stream processor.",
+			},
+			"stats": schema.StringAttribute{
```
Comment (AgustinBettati) [inline internal/service/streamprocessor/data_source_schema.go:44]: Interesting approach for including all stats. Just curious, does the API spec have the stats property fully typed, or left as a dynamic object (similar to pipeline)? would make sure to reflect this in the spec and ideally get input from the SP team.

**Review by AgustinBettati — APPROVED** [review]
LGTM with one doubt

File: internal/service/streamprocessor/data_source_schema.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/service/streamprocessor/data_source_schema.go:44]: Yes, same as the `pipeline`. Updated the spec 👍

**Comment by EspenAlbert** [comment]
merged. Can incorporate docs changes if they come up

File: internal/service/streamprocessor/data_source_schema.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				Required:            true,
+				Description:         "Human-readable label that identifies the stream processor.",
+				MarkdownDescription: "Human-readable label that identifies the stream processor.",
+			},
+			"project_id": schema.StringAttribute{
+				Required:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
```
Comment (kanchana-mongodb) [inline internal/service/streamprocessor/data_source_schema.go:37]: Should id be capitalized: ID?

**Review by kanchana-mongodb — COMMENTED** [review]
1 suggestions. LGTM

File: internal/service/streamprocessor/data_source_schema.go (hunk 3 above)
Comment (EspenAlbert) [reply internal/service/streamprocessor/data_source_schema.go:37]: `id` is used consistently throughout the API so I think we should keep it as is.

**Pull request merged**